pallets/logion_loc/src/lib.rs

```diff
@@ -200,13 +224,38 @@ pub mod pallet {
 				};
 				<LocMap<T>>::insert(loc_id, loc);
 
-				if <AccountLocsMap<T>>::contains_key(requester.clone()) {
-					<AccountLocsMap<T>>::mutate(requester.clone(), |accounts| {
-						let list = accounts.as_mut().unwrap();
-						list.push(loc_id.clone());
-					});
-				} else {
-					<AccountLocsMap<T>>::insert(requester.clone(), Vec::from([loc_id.clone()]));
+				match requester {
+					RequesterOf::<T>::None => if loc_type != LocType::Identity {
+						Err(Error::<T>::UnexpectedRequester)?
+					},
+					RequesterOf::<T>::Loc(requester_loc_id) => if loc_type != LocType::Transaction {
+						Err(Error::<T>::UnexpectedRequester)?
+					} else {
+						let requester_loc = <LocMap<T>>::get(&requester_loc_id);
+						match requester_loc {
+							None => Err(Error::<T>::UnexpectedRequester)?,
+							Some(loc) => if loc.loc_type != LocType::Identity {
+								Err(Error::<T>::UnexpectedRequester)?
```

Comment:
IMHO this should also fail when identity LOC:
- is not CLOSED.
- is VOID.

Reply:
Indeed!